### DISCUSSION TITLE
Rename NPM package to use organization prefix

### DISCUSSION THREAD

**Discussion by smellyshovel** [maplibre/maplibre-gl-js#2026] [Ideas]
There was an idea in Slack to rename the NPM package to use the name `@maplibre/maplibre-gl` instead of the current unprefixed `maplibre-gl`.

I'll put my 2 cents in as well. `maplibre-gl` (prefixed or unprefixed - doesn't matter) is a bit misleading. I think the package name should denote explicitly that it's a JS lib. So maybe also change name to `maplibre-gl-js` (`@maplibre/maplibre-gl-js`)?

Backwards compatibility note: would be nice to have both packages available still (prefixed and unprefixed). And not only available, but updated simultaneously as well. I believe the current CI/CD setup should allow to do so without any major changes.

Generally, I see it a nice idea to have all the MapLibre's sttuff located under one common scope.

**Comment by smellyshovel** [comment]
@JannikGM also noticed that the way one should install the lib is (almost) not mentioned in the docs. Whis should also be addressed.

**Comment by rotu** [comment]
I'd be in favor of dropping both "gl" and "js" - [js being the platform it runs on](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#name) and gl being an implementation detail.

**Reply by ghost** [reply]
I think the `gl` might also refer to "mapbox graphics language" (~ style definitions) instead of referring to OpenGL / WebGL.

I agree that the `js` is not necessary in the npm name / should be avoided.

**Reply by rotu** [reply]
[Here's what I found:](https://docs.mapbox.com/help/glossary/mapbox-gl/)

> Mapbox GL is a suite of open-source libraries for embedding customizable and responsive client-side maps in web, mobile, and desktop applications. Mapbox GL maps render at a high frame rate. The abbreviation "GL" comes from [OpenGL](http://en.wikipedia.org/wiki/OpenGL), the industry-standard Open Graphics Library.

So "GL" appears to be branding taken from OpenGL. It [does not appear](https://web.archive.org/web/20150905192650/https://www.mapbox.com/help/define-mapbox-gl/) to be related to the [style definitions](https://web.archive.org/web/20150905210222/https://www.mapbox.com/help/define-style/) (at least historically).

**Comment by wipfli** [comment]
Since v3 is coming closer, this could be a breaking change we want to consider... #1716

**Comment by HarelM** [comment]
This was discussed in the past (I'll probably won't be able to find where) and the decision was to keep the current name. I was also in favor of prefixing it with maplibre but this was not supported in a TSC meeting, I think.
Also, this is not an issue, converting it into a discussion...

**Comment by HarelM** [comment]
Found it! https://github.com/maplibre/maplibre-gl-js/issues/184
This was the original discussion.

**Reply by wipfli** [reply]
Thanks for finding the previous discussion. I guess this was some time ago and we can again discuss this change now...

**Reply by HarelM** [reply]
Has anything changed since this was discussed? Don't get me wrong, I'm all for it, but reopening decisions can be a "dangerous" path...

**Reply by wipfli** [reply]
I think the old argument that changing the name of the npm package leads to extra work for users without sufficient extra benefit is still valid actually...
